dtale/utils.py

```diff
@@ -733,3 +733,12 @@ def apply(df, func, *args, **kwargs):
         return df.swifter.progress_bar(False).apply(func, *args, **kwargs)
     except ImportError:
         return df.apply(func, *args, **kwargs)
+
+
+def optimize_df(df):
+    for col in df.select_dtypes(include=["object"]):
+        num_unique_values = len(df[col].unique())
+        num_total_values = len(df[col])
+        if num_unique_values / num_total_values < 0.5:
+            df[col] = df[col].astype("category")
```

Comment:
I was always told to use `.iloc`, but I heard differing things.  Do you know which is more correct?
```suggestion
            df.loc[:, col] = df[col].astype("category")
```

Reply:
I think they are the same?

Reply:
Thanks @Centropy-io 🙏 I remember pandas throwing warnings at one point in time if you didnt use iloc because of setting values on a copy, but in this instance it doesnt apply.